File: sdk/php/src/Interceptors/LoggingInterceptor.php
<?php

declare(strict_types=1);

namespace Velocity\SDK\Interceptors;

use Psr\Log\LoggerInterface;
use Psr\Log\NullLogger;

/**
 * Logs workflow lifecycle events using a PSR-3 logger.
 */
class LoggingInterceptor implements WorkflowInterceptorInterface
{
    private LoggerInterface $logger;
    private string $prefix;

    /**
     * @param LoggerInterface|null $logger PSR-3 logger; defaults to NullLogger.
     * @param string $prefix Log message prefix.
     */
    public function __construct(?LoggerInterface $logger = null, string $prefix = '[VELOCITY]')
    {
        $this->logger = $logger ?? new NullLogger();
        $this->prefix = $prefix;
    }

    /** @inheritDoc */
    public function onStart(string $workflowType, int $workflowKey): void
    {
        $this->logger->info("{$this->prefix} Workflow started: type={$workflowType}, key={$workflowKey}");
    }

    /** @inheritDoc */
    public function onComplete(int $workflowKey, string $result): void
    {
        $this->logger->info("{$this->prefix} Workflow completed: key={$workflowKey}");
    }

    /** @inheritDoc */
    public function onFail(int $workflowKey, \Throwable $error): void
    {
        $this->logger->error("{$this->prefix} Workflow failed: key={$workflowKey}, error={$error->getMessage()}");
    }

    /** @inheritDoc */
    public function onSignal(int $workflowKey, string $signalName): void
    {
        $this->logger->info("{$this->prefix} Workflow signal: key={$workflowKey}, signal={$signalName}");
    }
}

File: sdk/java/src/main/java/io/velocity/sdk/interceptors/LoggingInterceptor.java
package io.velocity.sdk.interceptors;

import java.util.logging.Logger;

/**
 * Logs workflow lifecycle events.
 */
public class LoggingInterceptor implements WorkflowInterceptor {

    private static final Logger logger = Logger.getLogger(LoggingInterceptor.class.getName());
    private final String prefix;

    /**
     * Create a new LoggingInterceptor with default prefix.
     */
    public LoggingInterceptor() {
        this("[VELOCITY]");
    }

    /**
     * Create a new LoggingInterceptor with a custom prefix.
     *
     * @param prefix log message prefix
     */
    public LoggingInterceptor(String prefix) {
        this.prefix = prefix;
    }

    @Override
    public void onStart(String workflowType, long workflowId) {
        logger.info(String.format("%s Workflow started: type=%s, id=%d", prefix, workflowType, workflowId));
    }

    @Override
    public void onComplete(long workflowId, byte[] result) {
        logger.info(String.format("%s Workflow completed: id=%d", prefix, workflowId));
    }

    @Override
    public void onFail(long workflowId, Throwable error) {
        logger.severe(String.format("%s Workflow failed: id=%d, error=%s", prefix, workflowId, error.getMessage()));
    }

    @Override
    public void onSignal(long workflowId, String signalName) {
        logger.info(String.format("%s Workflow signal: id=%d, signal=%s", prefix, workflowId, signalName));
    }
}
